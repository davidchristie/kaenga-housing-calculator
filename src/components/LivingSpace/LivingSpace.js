import React, { Component, PropTypes } from 'react'
import { Well } from 'react-bootstrap'

class LivingSpace extends Component {
  render () {
    const {
      total
    } = this.props
    return (
      <Well>
        <h1>Living space (including kitchens)</h1>
        <p>
          Total: <strong>{total}m<sup>2</sup></strong>
        </p>
      </Well>
    )
  }
}

LivingSpace.defaultProps = {
  total: 0
}

LivingSpace.propTypes = {
  total: PropTypes.number
}

export default LivingSpace
