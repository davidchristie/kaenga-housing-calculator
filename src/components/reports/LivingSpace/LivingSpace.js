import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Well } from 'react-bootstrap'

import Area from '../../units/Area'

class LivingSpace extends Component {
  render () {
    const {
      total
    } = this.props
    return (
      <Well>
        <h1>Living space</h1>
        <p>
          <span>Total (including kitchens): </span>
          <strong><Area value={total} /></strong>
        </p>
      </Well>
    )
  }
}

LivingSpace.defaultProps = {
  total: 0
}

LivingSpace.propTypes = {
  total: PropTypes.number
}

export default LivingSpace
