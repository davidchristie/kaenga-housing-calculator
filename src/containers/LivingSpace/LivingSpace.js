import { connect } from 'react-redux'

import LivingSpace from '../../components/LivingSpace'
import { getLivingSpace } from '../../selectors/living-space'

export default connect(
  state => {
    const livingSpace = getLivingSpace(state)
    return {
      ...livingSpace
    }
  }
)(LivingSpace)
