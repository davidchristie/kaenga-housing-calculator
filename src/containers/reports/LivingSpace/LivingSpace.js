import { connect } from 'react-redux'

import LivingSpace from '../../../components/reports/LivingSpace'
import { getLivingSpace } from '../../../selectors/living-space'

export default connect(
  state => {
    const livingSpace = getLivingSpace(state)
    return {
      ...livingSpace
    }
  }
)(LivingSpace)
