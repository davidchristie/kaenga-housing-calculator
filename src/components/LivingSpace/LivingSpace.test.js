import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import LivingSpace from './LivingSpace'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<LivingSpace />)).toMatchSnapshot()
})
