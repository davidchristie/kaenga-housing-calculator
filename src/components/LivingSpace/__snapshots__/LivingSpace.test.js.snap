exports[`test matches snapshot 1`] = `
<Well
  bsClass="well">
  <h1>
    Living space (including kitchens)
  </h1>
  <p>
    Total: 
    <strong>
      0
      m
      <sup>
        2
      </sup>
    </strong>
  </p>
</Well>
`;
